packages/pipeline/src/scripts/pull_ohlcv_cryptocompare.ts

```diff
@@ -0,0 +1,92 @@
+
+// convert these to env variables
+const MAX_CONCURRENT_REQUESTS = 18;
+const EARLIEST_BACKFILL_TIME = new Date('2010-09-01').getTime(); // the time when BTC/USD info starts appearing on Crypto Compare
+
+let connection: Connection;
+
+(async () => {
+    connection = await createConnection(ormConfig as ConnectionOptions);
+    const repository = connection.getRepository(OHLCVExternal);
+    const source = new CryptoCompareOHLCVSource(MAX_CONCURRENT_REQUESTS);
+
+    const tradingPairs = await fetchOHLCVTradingPairs(connection, SOURCE_NAME, EARLIEST_BACKFILL_TIME);
+    console.log(`Starting ${tradingPairs.length} job(s) to scrape Crypto Compare for OHLCV records...`);
+
+    const fetchAndSavePromises = tradingPairs.map(async pair => {
+        const pairs = source.generateBackfillIntervals(pair);
+        return fetchAndSaveAsync(source, repository, pairs);
+    });
+    await Promise.all(fetchAndSavePromises);
+    console.log(`Finished scraping OHLCV records from Crypto Compare, exiting...`);
+    process.exit(0);
+})().catch(handleError);
+
+async function fetchAndSaveAsync(
+    source: CryptoCompareOHLCVSource,
+    repository: Repository<OHLCVExternal>,
+    pairs: TradingPair[],
+): Promise<void> {
+    const sortAscTimestamp = (a: TradingPair, b: TradingPair): number => {
+        if (a.latest < b.latest) {
+            return -1;
+        } else if (a.latest > b.latest) {
+            return 1;
+        } else {
+            return 0;
+        }
+    };
+    pairs.sort(sortAscTimestamp);
+
+    let i = 0;
+    let shouldContinue = true;
+    while (i < pairs.length && shouldContinue) {
+        const p = pairs[i];
+        const rawRecords = await source.getAsync(p);
+
+        // need minimum 2 records to calculate startTime and endTime for each record
+        if (rawRecords.length < 2) {
+            i++;
+            continue;
+        }
+        const metadata: OHLCVMetadata = {
+            exchange: source.default_exchange,
+            fromSymbol: p.fromSymbol,
+            toSymbol: p.toSymbol,
+            source: SOURCE_NAME,
+            observedTimestamp: new Date().getTime(),
+        };
+        const parsedRecords = parseRecords(rawRecords, metadata);
+        try {
+            await saveRecordsAsync(repository, parsedRecords);
+            i++;
+        } catch (e) {
+            console.log(`Error saving OHLCVRecords, stopping task for ${JSON.stringify(p)} [${e}]`);
+            shouldContinue = false;
+        }
+    }
+    return Promise.resolve();
+}
+
+async function saveRecordsAsync(repository: Repository<OHLCVExternal>, records: OHLCVExternal[]): Promise<void> {
+    const metadata = [
+        records[0].fromSymbol,
+        records[0].toSymbol,
+        new Date(records[0].startTime),
+        new Date(records[records.length - 1].endTime),
+    ];
+
+    console.log(`Saving ${records.length} records to ${repository.metadata.name}... ${JSON.stringify(metadata)}`);
+    await repository.insert(records);
```

Comment:
1. Should we insert these records in batches instead of all at once? See [here for an example](https://github.com/0xProject/0x-monorepo/blob/73ad482e14b00fe37a33d931d781bcb083f11077/packages/pipeline/src/scripts/pull_competing_dex_trades.ts#L36).
2. Do we really mean to do an `insert` here instead of a `save`? The latter will handle primary key constraint violations gracefully but the former will not. Is there any scenario where we might get the same data more than once from subsequent script runs?

Reply:
1. The size of records is limited by the API call so I think we are safe.

2. Nope, just didn't know about `save`! Changing it to avoid unhandled errors, but also want to note that I'm not sure if primary key constraints apply in this table. We don't have any unique identifiers, unless we want to generate a composite key of all the primary columns (defined with typeorm annotations). And if we do have a duplicate of all those fields, we would probably want to ignore the dupes.